datafusion/optimizer/src/propagate_empty_relation.rs

```diff
@@ -94,29 +94,62 @@ impl OptimizerRule for PropagateEmptyRelation {
                 Ok(Transformed::no(LogicalPlan::CrossJoin(join.clone())))
             }
 
-            LogicalPlan::Join(ref join) if join.join_type == Inner => {
+            LogicalPlan::Join(ref join) => {
                 // TODO: For Join, more join type need to be careful:
```

Comment:
Does these comment need to remain?

Reply:
I removed only the ones I added, if we merge this I'll update the associated ticket.